auth_user_case_insensitive/tests/test_res_users.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2015-2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, registry
+from odoo.tests.common import TransactionCase
+from odoo.tools import mute_logger
+
+
+class TestResUsers(TransactionCase):
+
+    def setUp(self):
+        super(TestResUsers, self).setUp()
+        self.login = 'LasLabs@ExAmPlE.CoM'
+        self.partner_vals = {
+            'name': 'Partner',
+            'is_company': False,
+            'email': self.login,
+        }
+        self.vals = {
+            'name': 'User',
+            'login': self.login,
+            'password': 'password',
+        }
+        self.model_obj = self.env['res.users']
+
+    def _new_record(self):
+        """ It should enerate a new record to test with """
+        partner_id = self.env['res.partner'].create(self.partner_vals)
+        self.vals['partner_id'] = partner_id.id
+        return self.model_obj.create(self.vals)
+
+    def test_login_is_lowercased_on_create(self):
+        """ It should verify the login is set to lowercase on create """
+        rec_id = self._new_record()
+        self.assertEqual(
+            self.login.lower(), rec_id.login,
+            'Login was not lowercased when saved to db.',
+        )
+
+    def test_login_is_lowercased_on_write(self):
+        """ It should verify the login is set to lowercase on write """
+        rec_id = self._new_record()
+        rec_id.write({'login': self.login})
+        self.assertEqual(
+            self.login.lower(), rec_id.login,
+            'Login was not lowercased when saved to db.',
+        )
+
+    def test_login_login_is_lowercased(self):
+        """ It should verify the login is set to lowercase on login """
+        rec_id = self._new_record()
+        # We have to commit this cursor, because `_login` uses a fresh cursor
+        self.env.cr.commit()
```

Comment:
Oops... this passed unnoticed? A commit in a test is a clock bomb... :scream: :bomb: